FirebaseMLModelDownloader/Sources/ModelInfoRetriever.swift

```diff
@@ -109,9 +118,10 @@ extension ModelInfoRetriever {
   }
 
   /// Get model info from server.
-  func downloadModelInfo(completion: @escaping (DownloadError?) -> Void) {
+  func downloadModelInfo(completion: @escaping (Result<RemoteModelInfo?, DownloadError>) -> Void) {
```

Comment:
Yes, when the local model info *is* the latest model, the server returns a 304 (not modified) and no actual model info. In that case, the success case for ModelInfoRetriever returns nil to indicate to the caller to use the existing model info.

Reply:
`nil` looks like a pretty implicit indicator of the described case. Would it make sense to use a specific enum instead of just an optional to make the code more self-explaining, e.g.:
```
enum DownloadResult {
 case .notModified
 case .model( RemoteModelInfo)
}
```
In this case the method signature will be:
```
func downloadModelInfo(completion: @escaping (Result<DownloadResult, DownloadError>) -> Void)
```

Reply:
Good idea! Done.